Assets/HoloToolkit/Utilities/Scripts/Extensions/VectorExtensions.cs

```diff
@@ -50,5 +50,34 @@ public static Vector3 InverseTransformPoint(this Vector3 point, Vector3 translat
             var scaleInv = new Vector3(1 / lossyScale.x, 1 / lossyScale.y, 1 / lossyScale.z);
             return Vector3.Scale(scaleInv, (Quaternion.Inverse(rotation) * (point - translation)));
         }
+		
+		public static Vector3 Average(ICollection<Vector3> vectors)
+        {
+            if (vectors.Count == 0)
+            {
+                return Vector3.zero;
+            }
+
+            var x = 0f;
+            var y = 0f;
+            var z = 0f;
+            foreach (var pos in vectors)
```

Comment:
ICollection does not support random access so it's not possible to use 'for' instead of 'foreach'.
I suggest the more general 'IEnumerable' instead of 'ICollection'. Maybe one second overload for 'IList' where 'for' can be used.